spec/requests/memberships/approval_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe 'Membership approval', type: :request do
+  describe 'approval' do
+    let(:community) { create(:community) }
+
+    let(:not_moderator) { create(:person) }
+    let!(:membership_to_be_approve) { create(:membership, :guest, community: community, person: not_moderator) }
+
+    let(:moderator) { create(:person) }
+    let!(:moderator_membership) { create(:membership, community: community, person: moderator) }
+
+    let(:staff) { create(:person, staff: true) }
+    context 'staff' do
+      it 'returns http success' do
+        sign_in staff
+        post "/memberships/#{membership_to_be_approve.id}/approval"
+        expect(response).to redirect_to(community_memberships_path(membership_to_be_approve.community.slug))
```

Comment:
Not blocking: can we make an assertion that the transformation we want to occur actually happens?